src/coap_tinydtls.c

```diff
@@ -673,11 +673,16 @@ verify_ecdsa_key(struct dtls_context_t *dtls_context COAP_UNUSED,
   int ret;
 
   if (t_context && t_context->setup_data.validate_cn_call_back) {
+    coap_address_t remote_addr;
+    get_session_addr(dtls_session, &remote_addr);
+    coap_session_t *c_session = coap_session_get_by_peer(t_context->coap_context,
+                                                         &remote_addr, dtls_session->ifindex);
+    if (!c_session)
+      return -3;
+
     /* Need to build asn.1 certificate - code taken from tinydtls */
     uint8 *p;
-    uint8 buf[DTLS_CE_LENGTH];
-    coap_session_t *c_session;
-    coap_address_t remote_addr;
+    uint8 *buf = coap_malloc_type(COAP_DTLS_CONTEXT, DTLS_CE_LENGTH);
```

Comment:
COAP_DTLS_CONTEXT may be of fixed size (in particular LwIP) and so may not be >= DTLS_CE_LENGTH.

I would use COAP_STRING here.